### ISSUE TITLE
Moving cached rope frequencies to Spyre directly causes OOM

### ISSUE THREAD

**Issue by ani300** [torch-spyre/torch-spyre#418]
Due to current limitations in eager/inductor support for views, stack, and outer, the easiest way to compute the cached frequencies for Rope is to do it on cpu and then do this:

```py3
                self.cached_freqs[dev_idx][alpha] = torch.stack(
                    [
                        torch.cos(freqs),
                        -torch.sin(freqs),
                        torch.sin(freqs),
                        torch.cos(freqs),
                    ],
                    dim=2,
                ).view(*freqs.size(), 2, 2).to("spyre")
```

The issue is that this tensor is 131072x64x2x2, which is 64MB in DL16. When transferred to AIU, with the default settings (last dim is stick), it multiplies the size by 32, making it a 2GB tensor, which for some reason causes an OOM in the AIU memory, even though the allocator itself it's saying there's at least 96GB free. We should find out why that is happening, and also figure out if we can change the stick dimension either through making everything run properly on AIU or modifying the .to() function.
